Safety/Src/mpu6050_driver.cpp

```diff
@@ -0,0 +1,388 @@
+
+				data_check = 0x07;
+				HAL_I2C_Mem_Write(&hi2c1, IMU_ADDRESS, SMPLRT_DIV, 1, &data_check, 1, HAL_MAX_DELAY);
+
+				//Configure gyro to have range of +/- 250 dps
+				data_check = 0x00;
+				HAL_I2C_Mem_Write(&hi2c1, IMU_ADDRESS, GYRO_CONFIG, 1, &data_check, 1, HAL_MAX_DELAY);
+
+				//Configure Accelerometer to have range of +/- 2g
+				data_check = 0x00;
+				HAL_I2C_Mem_Write(&hi2c1, IMU_ADDRESS, ACCEL_CONFIG, 1, &data_check, 1, HAL_MAX_DELAY);
+			}
+
+	}
+
+}
+
+
+struct GyroData readGyro ()
+{
+		uint8_t data[6];
+		int16_t raw_x = 0, raw_y = 0, raw_z = 0;
+		struct GyroData gyro_res = {0};
+		HAL_StatusTypeDef status = HAL_OK;
+
+		status = HAL_I2C_Mem_Read(&hi2c1, IMU_ADDRESS, GYRO_XOUT_H, 1, data, 6, HAL_MAX_DELAY);
+
+		raw_x = (int16_t)(data[0] << 8 | data[1]);
+
+		raw_y = (int16_t)(data[2] << 8 | data[3]);
+
+		raw_z = (int16_t)(data[4] << 8 | data[5]);
+
+		/* Converting raw values into degrees per second.
+		 * Scaling by the sensitivity for the configured scale range.
+		 */
+
+
+		gyro_res.gyro_x = raw_x/kGryoCorrector;
+		gyro_res.gyro_y= raw_y/kGryoCorrector;
+		gyro_res.gyro_z= raw_z/kGryoCorrector;
+
+
+		return gyro_res;
+
+}
+
+
+struct AccelData readAccel ()
+{
+		uint8_t data[6];
+		int16_t raw_x = 0, raw_y = 0, raw_z = 0;
+
+		struct AccelData accel_res = {0};
+		HAL_StatusTypeDef status = HAL_OK;
+
+		status = HAL_I2C_Mem_Read(&hi2c1, IMU_ADDRESS, ACCEL_XOUT_H, 1, data, 6, HAL_MAX_DELAY);
+
+		raw_x = (int16_t)(data[0] << 8 | data[1]);
+
+		raw_y = (int16_t)(data[2] << 8 | data[3]);
+
+		raw_z = (int16_t)(data[4] << 8 | data[5]);
+
+		/* Converting raw values into acceleration in terms of g.
+		*  Scaling by the sensitivity for the configured scale range.
+		*/
+
+		accel_res.accel_x = raw_x/kAccelCorrector;
+		accel_res.accel_y = raw_y/kAccelCorrector;
+		accel_res.accel_z = raw_z/kAccelCorrector;
+
+
+		return accel_res;
+
+
+}
+
+/* USER CODE END 0 */
+
+/**
+  * @brief  The application entry point.
+  * @retval int
+  */
+int main(void)
+{
+  /* USER CODE BEGIN 1 */
+
+  /* USER CODE END 1 */
+
+  /* MCU Configuration--------------------------------------------------------*/
+
+  /* Reset of all peripherals, Initializes the Flash interface and the Systick. */
+  HAL_Init();
+
+  /* USER CODE BEGIN Init */
+
+  /* USER CODE END Init */
+
+  /* Configure the system clock */
+  SystemClock_Config();
+
+  /* USER CODE BEGIN SysInit */
+
+  /* USER CODE END SysInit */
+
+  /* Initialize all configured peripherals */
+  MX_GPIO_Init();
+  MX_I2C1_Init();
+  /* USER CODE BEGIN 2 */
+
+  imuInit();
+
+//  checkReg();
+  /* USER CODE END 2 */
+
+  /* Infinite loop */
+  /* USER CODE BEGIN WHILE */
+  while (1)
+  {
+
+	  //read_gyro here
+    /* USER CODE END WHILE */
+
+    /* USER CODE BEGIN 3 */
+	struct GyroData gyro_result = readGyro();
+	struct AccelData accel_result = readAccel();
+	HAL_Delay(10);
+
+
+
+  }
+  /* USER CODE END 3 */
+}
```

Comment:
As Dhruv pointed out, since the driver works we should now make it mirror the design pattern used in Autopilot. 

The final product will have a GetResult() function which will end up doing a similar thing here :))